erpnext/manufacturing/doctype/bom/bom_tree.js

```diff
@@ -16,7 +16,12 @@ frappe.treeview_settings["BOM"] = {
 	show_expand_all: false,
 	get_label: function (node) {
 		if (node.data.qty) {
-			return node.data.qty + " x " + node.data.item_code;
+			return `
+			<span style="font-weight: bold;">
+				${node.data.item_code}: ${node.data.item_name}
+			</span>
+			<span> — [${node.data.qty} ${node.data.stock_uom}]</span>
+			`;
```

Comment:
```suggestion
			let label = frappe.utils.escape_html(node.data.item_code);
			if (node.data.item_name && node.data.item_code !== node.data.item_name) {
				label += `: ${frappe.utils.escape_html(node.data.item_name)}`;
			}
			return `${label} <span class="badge badge-pill badge-light">${
				node.data.qty
			} ${frappe.utils.escape_html(node.data.stock_uom)}</span>`;
```

<img width="955" height="472" alt="Image" src="https://github.com/user-attachments/assets/40d2525c-c7be-4e85-a98c-1949c38468a8" />